File: version/1.8.9/src/main/java/com/github/sorusclient/client/feature/impl/enhancements/v1_8_9/EnhancementsHook.java
/*
 * Copyright (c) 2022. Danterus
 * Copyright (c) 2022. Sorus Contributors
 *
 * SPDX-License-Identifier: MPL-2.0
 */

package com.github.sorusclient.client.feature.impl.enhancements.v1_8_9;

import com.github.sorusclient.client.feature.impl.enhancements.Enhancements;
import v1_8_9.com.mojang.blaze3d.platform.GlStateManager;
import v1_8_9.net.minecraft.client.MinecraftClient;
import v1_8_9.org.lwjgl.opengl.Display;

@SuppressWarnings("unused")
public class EnhancementsHook {

    public static Object options;

    public static void preRenderFireFirstPerson() {
        GlStateManager.pushMatrix();
        GlStateManager.translated(0.0D, -Enhancements.INSTANCE.getFireHeightValue() * 0.4D, 0.0D);
    }

    public static void postRenderFireFirstPerson() {
        GlStateManager.popMatrix();
    }

    public static int modifyPotionOffset(int offset) {
        return com.github.sorusclient.client.feature.impl.enhancements.v1_8_9.Enhancements.INSTANCE.isCenteredInventoryValue() ? 0 : offset;
    }

    public static float modifyPitch(float pitch) {
        return MinecraftClient.getInstance().options.perspective == 2 ? -pitch : pitch;
    }

    public static float modifyPrevPitch(float prevPitch) {
        return MinecraftClient.getInstance().options.perspective == 2 ? -prevPitch : prevPitch;
    }

    public static void onSave() {
        Enhancements.INSTANCE.saveSettings();
    }

    public static void onLoad(Object options) {
        EnhancementsHook.options = options;
        Enhancements.INSTANCE.loadSettings();
    }

    public static void onStop() {
        Display.destroy();
    }

    private EnhancementsHook() {
    }
}

File: version/1.8.9/src/main/kotlin/com/github/sorusclient/client/feature/impl/enhancements/v1_8_9/EnhancementsHook.kt
/*
 * Copyright (c) 2022. Danterus
 * Copyright (c) 2022. Sorus Contributors
 *
 * SPDX-License-Identifier: MPL-2.0
 */

package com.github.sorusclient.client.feature.impl.enhancements.v1_8_9

import com.github.sorusclient.client.feature.impl.enhancements.Enhancements
import v1_8_9.com.mojang.blaze3d.platform.GlStateManager
import v1_8_9.net.minecraft.client.MinecraftClient
import v1_8_9.net.minecraft.client.gui.screen.Screen
import v1_8_9.net.minecraft.client.gui.screen.ingame.CreativeInventoryScreen
import v1_8_9.net.minecraft.client.gui.screen.ingame.SurvivalInventoryScreen
import v1_8_9.net.minecraft.client.options.KeyBinding
import v1_8_9.org.lwjgl.input.Keyboard
import v1_8_9.org.lwjgl.opengl.Display

@Suppress("UNUSED")
object EnhancementsHook {

    @JvmStatic
    fun preRenderFireFirstPerson() {
        GlStateManager.pushMatrix()
        GlStateManager.translated(0.0, -Enhancements.getFireHeightValue() * 0.4, 0.0)
    }

    @JvmStatic
    fun postRenderFireFirstPerson() {
        GlStateManager.popMatrix()
    }

    @JvmStatic
    fun modifyPotionOffset(offset: Int): Int {
        return if (com.github.sorusclient.client.feature.impl.enhancements.v1_8_9.Enhancements.isCenteredInventoryValue()) {
            0
        } else offset
    }

    @JvmStatic
    fun modifyPitch(pitch: Float): Float {
        return if (MinecraftClient.getInstance().options.perspective == 2) {
            -pitch
        } else pitch
    }

    @JvmStatic
    fun modifyPrevPitch(prevPitch: Float): Float {
        return if (MinecraftClient.getInstance().options.perspective == 2) {
            -prevPitch
        } else prevPitch
    }

    @JvmStatic
    fun onSave() {
        Enhancements.saveSettings()
    }

    lateinit var options: Any

    @JvmStatic
    fun onLoad(options: Any) {
        EnhancementsHook.options = options
        Enhancements.loadSettings()
    }

    @JvmStatic
    fun onStop() {
        Display.destroy()
    }

    @JvmStatic
    fun modifyBobView(bobView: Boolean): Boolean {
        if (Enhancements.isPartialViewBobbing()) {
            return false
        }

        return bobView
    }

    @JvmStatic
    fun onCloseContainer(screen: Screen?) {
        if (screen is CreativeInventoryScreen) return

        for (keyBind in MinecraftClient.getInstance().options.keysAll) {
            if (keyBind.code > 0 && Keyboard.isKeyDown(keyBind.code) && keyBind != MinecraftClient.getInstance().options.keyInventory) {
                KeyBinding.setKeyPressed(keyBind.code, true)
            }
        }
    }

}